### PR TITLE
add microservice for intent detection

### PR DESCRIPTION
**Pull request by Liangyx2**
## Description

Added Intent Detection Microservice by TGI

## Issues

n/a.

## Type of change

- [x] New feature (non-breaking change which adds new functionality)

## Dependencies

docarray[full]
fastapi
huggingface_hub
langchain==0.1.16
opentelemetry-api
opentelemetry-exporter-otlp
opentelemetry-sdk
shortuuid

## Tests

test each changed functions.

### CODE REVIEW

**Comment by chensuyue** [comment]
Please contribute an e2e test for this microservice, like this one: https://github.com/opea-project/GenAIComps/blob/main/tests/test_reranks.sh, please name it as `test_intent_detection.sh`, I will let GHA trigger the test.

File: comps/intent_detection/llm.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        top_k=input.top_k,
+        top_p=input.top_p,
+        typical_p=input.typical_p,
+        temperature=input.temperature,
+        repetition_penalty=input.repetition_penalty,
+        streaming=input.streaming,
+        timeout=600,
+    )
+
+    intent_template = """Please identify the intent of the user query. You may only respond with "chitchat" or "QA" without explanations or engaging in conversation.
```
Comment (lkk12014402) [inline comps/intent_detection/llm.py:47]: wouldn't it be better to create a `template.py` file to hold the template?

File: comps/intent_detection/llm.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    ### User Query: {query}, ### Response: """
+
+    prompt = PromptTemplate(template=intent_template, input_variables=["query"])
```
Comment (lkk12014402) [inline comps/intent_detection/llm.py:50]: If the code is universal, can it be moved externally without initializing it for every request?

File: comps/intent_detection/llm.py (hunk 1 above)
Comment (Liangyx2) [reply comps/intent_detection/llm.py]: yes, I add template.py.

**Review by lkk12014402 — APPROVED** [review]

File: comps/intent_detection/langchain/intent_detection.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+from comps import GeneratedDoc, LLMParamsDoc, ServiceType, opea_microservices, register_microservice
+from comps.intent_detection.langchain.template import IntentTemplate
+
+
+@register_microservice(
+    name="opea_service@llm_intent",
+    service_type=ServiceType.LLM,
+    endpoint="/v1/chat/intent",
+    host="0.0.0.0",
+    port=9000,
```
Comment (lvliang-intel) [inline comps/intent_detection/langchain/intent_detection.py:19]: don't use port 9000 since it's occupied by llm microservice.

Comment (Liangyx2) [reply comps/intent_detection/langchain/intent_detection.py:19]: which port do you recommend? I will change.

**Review by lvliang-intel — APPROVED** [review]

**Pull request merged**